databroker/core.py

```diff
@@ -1507,6 +1507,134 @@ def resource(self, doc):
         self.resources[doc['uid']] = doc
 
 
+class SingleRunCache:
+    """
+    Collect the document from one Run and, when complete, provide a BlueskyRun.
+
+    Parameters
+    ----------
+    handler_registry: dict, optional
+        This is passed to the Filler or whatever class is given in the
+        filler_class parameter below.
+
+        Maps each 'spec' (a string identifying a given type or external
+        resource) to a handler class.
+
+        A 'handler class' may be any callable with the signature::
+
+            handler_class(resource_path, root, **resource_kwargs)
+
+        It is expected to return an object, a 'handler instance', which is also
+        callable and has the following signature::
+
+            handler_instance(**datum_kwargs)
+
+        As the names 'handler class' and 'handler instance' suggest, this is
+        typically implemented using a class that implements ``__init__`` and
+        ``__call__``, with the respective signatures. But in general it may be
+        any callable-that-returns-a-callable.
+    root_map: dict, optional
+        This is passed to Filler or whatever class is given in the filler_class
+        parameter below.
+
+        str -> str mapping to account for temporarily moved/copied/remounted
+        files.  Any resources which have a ``root`` in ``root_map`` will be
+        loaded using the mapped ``root``.
+    filler_class: type
```

Comment:
That was my intent, but it could be made more clear. This expects a class like `Filler` or `NoFiller`, and `type(Filler)` (or any class) is of course `type`.

Reply:
OK, let's leave it as is.